pkg/storaged/utils.js

```diff
@@ -843,7 +865,15 @@ export function get_active_usage(client, path, top_action, child_action, is_temp
             });
         }
 
-        if (fsys && fsys.MountPoints.length > 0) {
+        // HACK: This works for unmounting a btrfs subvolume, however on trying
+        // to mount a second btrfs volume this will try to unmount the block
+        // device where btrfs is on. So we just check if there is a mount_point.
+        // Note that we don't handle any of the overmounting stuff now.
+        if (btrfs_volume && subvol) {
+            if (subvol.mount_point) {
+                enter_unmount(block, subvol.mount_point, true);
+            }
+        } else if (fsys && fsys.MountPoints.length > 0) {
```

Comment:
@mvollmer would love to have a chat about this part of the code. This now successfully works for mounting/unmounting of subvolumes but can't tell if this is really correct.

Reply:
(this was discussed on IRC with lotsa details. summary: almost correct. :-)

Reply:
As discussed on IRC :-)

- subvol should include the current MountPoints (already collected by findmnt --json)
- using those mountpoints, use the same logic for unmounting.